code/modules/projectiles/updated_projectiles/projectile.dm

```diff
@@ -627,7 +635,12 @@ Normal range for a defender's bullet resist should be something around 30-50. ~N
 			#endif
 
 		var/penetration = P.ammo.penetration > 0 || armor > 0 ? P.ammo.penetration : 0
+		if(src == P.shot_from.sniper_target(src))
```

Comment:
This should be handled in the gun, not in defense.

Reply:
How would you make it do more damage from the gun-side specifically without having unintended effects... like say doing way more damage than you should to a wall (which is why I set it up like this)?